nix/default.nix

```diff
@@ -30,7 +30,7 @@
   hidpiXWayland ? false,
   legacyRenderer ? false,
   nvidiaPatches ? false,
-  withSystemd ? true,
+  withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd,
```

Comment:
unless this change is relevant to the separation of wayland-scanner from wayland, it belongs in a separate PR

Reply:
no directly relevant, here is the separate PR：https://github.com/hyprwm/Hyprland/pull/2823